module/Api/src/Entity/Messaging/AbstractMessagingUserMessageRead.php

```diff
@@ -176,6 +189,37 @@ public function getLastModifiedBy()
         return $this->lastModifiedBy;
     }
 
+    /**
+     * Set the last read on
+     *
+     * @param \DateTime $lastReadOn new value being set
+     *
+     * @return MessagingUserMessageRead
+     */
+    public function setLastReadOn($lastReadOn)
+    {
+        $this->lastReadOn = $lastReadOn;
+
+        return $this;
+    }
+
+    /**
+     * Get the last read on
+     *
+     * @param bool $asDateTime If true will always return a \DateTime (or null) never a string datetime
+     *
+     * @return \DateTime|string
+
+     */
+    public function getLastReadOn($asDateTime = false)
+    {
+        if ($asDateTime === true) {
+            return $this->asDateTime($this->lastReadOn);
+        }
+
+        return $this->lastReadOn;
```

Comment:
If you setLastReadOn(DateTime object), then call getLastReadOn() you'll get a DateTime object and not a string.